SU2_CFD/src/variables/CDiscAdjVariable.cpp

```diff
@@ -37,129 +37,44 @@
 
 #include "../../include/variables/CDiscAdjVariable.hpp"
 
-CDiscAdjVariable::CDiscAdjVariable() : CVariable() {
 
-  /*--- Initialize arrays to NULL ---*/
-
-  Solution_Direct = NULL;
-  Sensitivity     = NULL;
-
-  DualTime_Derivative   = NULL;
-  DualTime_Derivative_n = NULL;
-
-  Geometry_Direct       = NULL;
-  Solution_Geometry     = NULL;
-  Solution_Geometry_Old = NULL;
-  Cross_Term_Derivative = NULL;
-
-  Solution_BGS            = NULL;
-  Solution_BGS_k          = NULL;
-  Solution_Geometry_BGS_k = NULL;
-
-  Geometry_CrossTerm_Derivative      = NULL;
-  Geometry_CrossTerm_Derivative_Flow = NULL;
-
-}
-
-CDiscAdjVariable::CDiscAdjVariable(su2double* val_solution, unsigned short val_ndim, unsigned short val_nvar,
-                                   CConfig *config) : CVariable(val_ndim, val_nvar, config) {
-
-  bool dual_time = (config->GetTime_Marching() == DT_STEPPING_1ST)
-      || (config->GetTime_Marching() == DT_STEPPING_2ND);
+CDiscAdjVariable::CDiscAdjVariable(const su2double* sol, Idx_t npoint, Idx_t ndim, Idx_t nvar, CConfig *config)
```

Comment:
so cleeeaaannn

Reply:
Thanks, I was also surprised when I started cleaning up constructors at how much could be removed, and no risk of memory leaks now...